File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.java
class Solution {
    public int solution(int[] num_list) {
        // 정답 선언 및 초기화
        int answer = 0;
        // 홀수와 짝수를 더하지 않고 이어붙일 String 선언 및 초기화 - 이어붙인 후에 형변환
        String even = "";
        String odd = "";
        
        // num_list의 홀수와 짝수를 분리할 for 문
        for(int i=0; i<num_list.length; i++){
            if(num_list[i]%2 == 0){
                even += num_list[i] + ""; 
            } else {
                odd += num_list[i] + "";
            }
        }
        // 짝수와 홀수를 이어붙인 문자열을 형변환해 더하기
        answer = Integer.valueOf(even) + Integer.valueOf(odd);
        
        // 정답 리턴
        return answer;
    }
}

File: 프로그래머스/0/181928. 이어 붙인 수/이어 붙인 수.py
def solution(num_list):
    answer = 0
    even = ''
    odd = ''
    i = 0
    
    while i < len(num_list):
        if num_list[i] % 2 == 0:
            even += str(num_list[i])
            i += 1
        else:
            odd += str(num_list[i])
            i += 1
    
    answer = int(even) + int(odd)
    
    return answer